libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/line/api/VanishingRouteLineTest.kt

```diff
@@ -70,6 +70,29 @@ class VanishingRouteLineTest {
         )
     }
 
+    @Test
+    fun initWithRoute_clearState() = coroutineRule.runBlockingTest {
+        val route = getRoute()
+        val vanishingRouteLine = VanishingRouteLine()
+
+        vanishingRouteLine.initWithRoute(route)
+        assertEquals(
+            8,
+            vanishingRouteLine.primaryRouteLineGranularDistances!!.distancesArray.size()
+        )
+        unmockkObject(ThreadController)
+        val jobControl = ThreadController.getMainScopeAndRootJob()
+        mockkObject(ThreadController)
+
+        every { ThreadController.getMainScopeAndRootJob() } returns jobControl
+        Runnable {
+            vanishingRouteLine.initWithRoute(getVeryLongRoute())
+        }.run()
+        jobControl.job.cancel()
+
+        assertNull(vanishingRouteLine.primaryRouteLineGranularDistances)
+    }
```

Comment:
Should we add `unmockkObject(ThreadController)` at the end so we know for sure there are no leftovers potentially impacting other tests?

Reply:
the teardown method has unmockkObject(ThreadController)